posts/2019-01-22-the-5-raisons-detre-of-testing.md

```diff
@@ -0,0 +1,219 @@
+---
+title: The 5 Raisons d'Être of Testing
+---
+
+<style>
+.story {
+  background-color: lightblue;
+  padding: 20px;
+  border-radius: 10px;
+}
+
+</style>
+
+# The 5 Raisons d'Être of Testing
+
+I describe five of the reasons why I think tests are important:
+
+* Ensure that the code you've written works.
+* Ensure that the code you've written works tomorrow.
+* Ensure that the code you've written works elsewhere.
+* Spot interface issues early on.
+* Show how your feature is supposed to work, and how it should be used.
+
+**important**: None of the animals used in the examples were harmed in the
+making of this post.
+
+---
+
+First, let's get in the mood (or [skip
+ahead](#ensure-that-the-code-youve-written-works)):
+
+<div class="story">
+
+You are a backend engineer at `MyCuteThings.com`, the famous website of cute
+things. Your boss comes to you and Tommy (your frontend sidekick) one Wednesday
+morning and declares:
+
+> There shall be a `"breed"` dropdown listing two cat breeds: Persian and
+> Siamese. When the user selects a breed in `"breed"` a picture of a cat of
+> that breed shall be displayed to the user. Feature codename: "ninja".
+>
+> -- your boss
+
+Tommy will deal with the HTML, CSS, and whatever else frontend people use. Your
+job is to provide an API endpoint that will return the bytes of the correct
+picture. You find a picture of a very cute Persian cat that you save as
+`cat.png`. You take a picture of your very own Siamese cat that you save as
+`CAT.png`. You settle on the following REST call:
```

Comment:
the filenames look suspicious, waiting for the non-case sensitive filesystem like macOS' or NTFS